src/nti/fakestatsd/__init__.py

```diff
@@ -91,6 +91,11 @@ def __init__(self, prefix=''):
 
         class DummySocket(object):
             def sendto(self, data, addr):
+                # The client encoded to bytes
+                assert isinstance(data, bytes)
+                # We always want native strings here, that's what the
+                # user specified when calling the StatsdClient methods.
+                data = data.decode('utf-8') if bytes is not str else data
```

Comment:
One thing worth noting here is that perfmetrics encodes using [ascii](https://github.com/hathawsh/perfmetrics/blob/master/perfmetrics/statsd.py#L81). We probably want to match that encoding here.

Reply:
UTF-8 is a superset of ASCII---if it can be encoded using ASCII, it can be decoded to an identical string using UTF-8. 

UTF-8 is the "standard" encoding. If I see `val.decode("anything other than utf-8")` I have to stop and ask: Why is that not using UTF-8? If there's not a comment right there, I have to do some digging. Even if there is a comment, I still have to read and think about it. Using UTF-8 essentially makes the decoding invisible.

So I would prefer to keep it as UTF-8, but I can understand the argument about matching. If you feel strongly I can change it.